plugin/src/main/java/io/jenkins/plugins/casc/ConfigurationAsCode.java

```diff
@@ -494,14 +494,17 @@ public Node toYaml(CNode config) throws ConfiguratorException {
 
                 final DumperOptions.ScalarStyle style = scalar.isRaw() ? PLAIN : DOUBLE_QUOTED;
 
-                return new ScalarNode(getTag(scalar.getFormat()), value, null, null, style);
+                return new ScalarNode(getTag(scalar), value, null, null, style);
         }
     }
 
-    private Tag getTag(Scalar.Format format) {
-        switch (format) {
+    private Tag getTag(Scalar scalar) {
+        switch (scalar.getFormat()) {
             case NUMBER:
-                return Tag.INT;
+                if (scalar.getValue().matches("^-?[0-9]+$"))
```

Comment:
would not be easier to match on anything with `.` in it?

Reply:
I thought of that, but I'm unsure if large values ever get stringified to scientific form, eg `1.3e+5` or `3e-9` (notice how the latter is a valid float that does not contain a `.`).

Reply:
Perhaps better to add a Float/Double Format as I did for multiline? Rather than checking the value based on regex?

I'd even suggest adding the following:

```
    public Scalar(double instance) {
        this.value = String.valueOf(instance);
        this.format = Format.FLOATING;
        this.raw = true;
    }

    public Scalar(float instance) {
        this.value = String.valueOf(instance);
        this.format = Format.FLOATING;
        this.raw = true;
    }
```

Reply:
Yeah, adding a `FLOATING` format looks nicer to me. Should I make the change in this PR?

Reply:
Please do :)

Reply:
I added some `instanceof` tests in the existing constructor taking `Number` instead of adding overloads for `Float` and `Double` because overload resolution would sometimes pick up the one you wouldn't expect.